microsetta_admin/templates/add_barcode_to_kit.html

```diff
@@ -0,0 +1,160 @@
+                num_samples: "required",
+                user_barcode: {
+                    barcodePattern: true
+                },
+
+                // Make sure the form is submitted to the destination defined
+                // in the "action" attribute of the form when valid
+                submitHandler: function (form) {
+                    form.submit();
+                }
+            }
+        });
+
+        // Handle form submission
+        $("form[name='kit_form']").submit(function(event) {
+            // Check which button was clicked
+            var submitButton = $("input[type='submit'][clicked=true]").attr('name');
+
+            if (submitButton == 'upload') {
+                // Prevent default form submission
+                event.preventDefault();
+                // Manually submit the form
+                $("#kit_form").ajaxSubmit({
+                    success: function(response) {
+                        // Handle success response
+                        console.log(response);
+                    },
+                    error: function(xhr, status, error) {
+                        // Handle error response
+                        console.error(error);
+                    }
+                });
+            }
+        });
+
+        // Track which button was clicked
+        $("input[type='submit']").click(function() {
+            $("input[type='submit']", $(this).parents("form")).removeAttr("clicked");
+            $(this).attr("clicked", "true");
+        });
+    });
+
+    function setBarcodeFromList() {
+        var barcodeList = document.getElementById("barcodeList");
+        if (barcodeList && barcodeList.children.length > 0) {
+            var barcode = barcodeList.children[0].innerText;
+            document.getElementById("user_barcode").value = barcode;
+        }
+        else {
+            alert("No barcodes to generate");
+        }
+    }
+
+</script>
+
+{% endblock %}
+{% block content %}
+<h3>Add Barcode to Kit(s)</h3>
+<style>
+    select {
+        width: 250px;
+        margin: 10px;
+    }
+</style>
+<div style="height: 335px;">
+    {% if error_message %}
+        <p style="color:red">
+            {{ error_message |e }}
+        </p>
+    {% endif %}
+
+    <form name="kit_form" id="kit_form" method="POST" enctype="multipart/form-data">
+        <table>
+            <tr>
+                <td><label for="project_id">Project ID: </label></td>
+                <td><input type="number" name="project_id" id="project_id"></td>
+            </tr>
+            <tr>
+                <td><label for="num_kits">Number of kit(s): </label></td>
+                <td><input type="number" name="num_kits" id="num_kits" min="1"></td>
+            </tr>
+            <tr>
+                <td><label for="num_samples">Number of sample(s) per kit: </label></td>
+                <td><input type="number" name="num_samples" id="num_samples" min="1"></td>
+            </tr>
+            <tr>
```

Comment:
For visual clarity, let's start the process by having the user click one of two buttons:
1) Add Barcode to Single Kit
2) Add Barcodes to Multiple Kits

Depending on which button they click, they'll then see the appropriate form.
Single Kit:
- Text input for Kit ID (required)
- Text input for Barcode (conditionally required if Generate Barcode is not checked)
- Checkbox for Generate Barcode (conditionally required if Barcode is not provided)
- Add Barcode button (submit)

Multiple Kits:
- File upload input for Kit IDs (required)
- File upload input for Barcodes (conditionally required if Generate Barcodes is not checked)
- Checkbox for Generate Barcodes (conditionally required if Barcodes are not provided)
- Add Barcodes button (submit)